alpha/apps/kaltura/lib/kCategoryEventHandler.php

```diff
@@ -0,0 +1,305 @@
+<?php
+class kCategoryEventHandler implements kObjectDeletedEventConsumer, kObjectCreatedEventConsumer, kObjectChangedEventConsumer
+{
+	
+	const PUBLIC_AGGREGATION_CATEGORY = 'publicAggregationCategory';
+	/* (non-PHPdoc)
+	 * @see kObjectChangedEventConsumer::objectChanged()
+	 */
+	public function objectChanged(BaseObject $object, array $modifiedColumns) {
+		if ($object instanceof categoryEntry)
+		{
+			$this->handleCategoryEntryCreated($object);
+		}
+		
+		if ($object instanceof category)
+		{
+			$this->handleCategoryChanged($object, $modifiedColumns);
+		}
+		
+		return true;
+	}
+	
+	protected function handleCategoryChanged (category $object, array $modifiedColumns)
+	{
+		$oldCustomDataValues = $object->getCustomDataOldValues();
+		$oldAggregationCategories = isset ($oldCustomDataValues[''][category::AGGREGATION_CATEGORIES]) ? explode (',', $oldCustomDataValues[''][category::AGGREGATION_CATEGORIES]): array();
+		$currentAggregationCategories = explode (',', $object->getAggregationCategories());
+		
+		$aggregationCategoriesToAdd = array_diff($currentAggregationCategories, $oldAggregationCategories);
+		KalturaLog::info ("Copying entries from category ID [" . $object->getId() . "] to aggregation channels: " . print_r($aggregationCategoriesToAdd, true));
+		$this->addToAggregationCategories($object, $aggregationCategoriesToAdd);
+		
+		$aggregationCategoriesToRemove = array_diff ($oldAggregationCategories, $currentAggregationCategories);
+		
+		KalturaLog::info ("Removing entries from category ID [" . $object->getId() . "] to aggregation channels: " . print_r($aggregationCategoriesToRemove, true));
+		$this->deleteFromAggregationChannels ($object, $aggregationCategoriesToRemove);
+		
+	}
+	
+	protected function deleteFromAggregationChannels (category $object, array $aggregationCatIds)
+	{
+		$aggregationCategories = categoryPeer::retrieveByPKs($aggregationCatIds);
+		foreach ($aggregationCategories as $aggregationCategory)
+		{
+			/* @var $aggregationCategory category */
+			$currentPublishingCategories = explode (',', $aggregationCategory->getPublishingCategories());
+			$currentPublishingCategories = array_diff($currentPublishingCategories, array ($object->getId()));
+			$aggregationCategory->setPublishingCategories(implode(',', $currentPublishingCategories));
+			$aggregationCategory->save();
+			
+			$this->addDeleteAggregationCategoryEntryJob ($object, $aggregationCategory);
+		}
+	}
+	
+	protected function addDeleteAggregationCategoryEntryJob (category $object, category $aggregationCategory)
+	{
+		$filter = new entryFilter();
+		$filter->set("_matchand_categories_ids", $object->getId().','.$aggregationCategory->getId());
+		$filter->set ("_notcontains_categories_ids", $aggregationCategory->getPublishingCategories());
+		
+		$additionalParameters = array();
+		$pair[self::PUBLIC_AGGREGATION_CATEGORY] = $aggregationCategory->getId();
```

Comment:
I think that you should initialize `$pair` first (as an array), otherwise it could throw a PHP notice